CONTRIBUTING.md

```diff
@@ -2,32 +2,97 @@
 
-## Peer Review
+Next, you'll need to clone `username/sandbox`, which downloads a copy to your computer. In a terminal, run:
 
-All pull requests undergo peer review in which project participants review the changes. Reviewers may suggest modifications or even note their disapproval. Reviewers can note their approval by commenting along the lines of ":+1:", "looks good to me", or "approved". As a reviewer, it's helpful to note the type of review you performed: did you test the code, look it over, or are you just supporting the concept?
+```sh
+# Clone your fork locally
+git clone https://github.com/username/sandbox
 
-Before a repository maintainer merges a pull request, there must be at least one affirmative review. If there is any unaddressed criticism or disapproval, a repository maintainer will determine how to proceed and may wait for additional feedback.
+# Change directory into the new repository
+cd sandbox
 
-**Happy contributing!**
+# Add the original cognoma repository as a remote repository named upstream
+git remote add upstream https://github.com/cognoma/sandbox.git
+```
+
+The final command makes git aware of `cognoma/sandbox`, so you can retrieve upstream changes in the future by running `git fetch upstream`.
+
+Now you have a local directory of the git repository. Any changes to the contents of the directory can now be tracked using git. But before you make your desired change, we recommend creating a branch. When you clone a repository, you will, by default, be on the `master` branch. However, you don't want to make changes on your `master` branch, since it's best to keep it synced with the `upstream` repository. Therefore, you will create a new branch for making your change. So say I wanted to add my name to `cognoma/sandox`, I would create a new branch named `add-username` with the following command:
+
+```sh
+# Create a new branch named add-username and switch to it
+git checkout --branch add-username
+```
+
+Now you will be on your `add-username` branch. You should now make your change, which in this example will be adding a new file but can also be modifying an existing file. Let's say you create the file `people/username.md` in your text editor of choice. Now you will have to stage your change by running:
+
+```sh
+# Show what changes git detects.
+# Not essential but good practice as it may alert you to an issue.
+git status
+
+# Add the file you've created
+git add people/username.md
+```
 
-***
+Next you will have to make a commit, which takes a snapshot of everything that's been staged (using `git add` above). To commit, run `git commit`. A text editor should appear that allows you to write a [commit message](https://chris.beams.io/posts/git-commit/) where you should describe the changes in the commit. If all succeeds, you will have added to the git repository and your commit will be visible when you run `git log`.
```

Comment:
can also probably add instructions for `git commit -m` functionality explicitly in bash snippet